core/src/main/java/hudson/model/listeners/RunListener.java

```diff
@@ -207,6 +216,21 @@ public static void fireCompleted(Run r, @Nonnull TaskListener listener) {
     }
 
     /**
+     * Fires the {@link #onInitialize(Run)} event.
+     */
+    public static void fireInitialize(Run r) {
```

Comment:
shouldn't something be calling this?  (links to my prior question - what does it mean for a Run to be initialized - when does this happen) - are you expecting all slave types to have to fire this?